xep-0388.xml

```diff
@@ -686,131 +695,141 @@
+    <xs:complexType>
+      <xs:sequence>
+        <xs:any namespace='##other' minOccurs='0' maxOccurs='unbounded' processContents='lax'/>
+        <xs:element name="additional-data" type="SaslData" minOccurs="0"/>
+        <xs:element name="authorization-identifier" type="Jid"/>
+        <xs:any namespace='##other' minOccurs='0' maxOccurs='unbounded' processContents='lax'/>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="failure">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element name="text" type="xs:string" minOccurs="0"/>
+        <xs:any namespace='##other' minOccurs='0' maxOccurs='unbounded' processContents='lax'/>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="continue">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element name="additional-data" type="SaslData"/>
+        <xs:element name="tasks">
+          <xs:complexType>
             <xs:sequence>
-                <xs:any/>
+              <xs:element name="task" type="SaslMechName" maxOccurs='unbounded'/>
             </xs:sequence>
-        </xs:complexType>
-    </xs:element>
-
-    <xs:simpleType name="Jid">
-        <xs:restriction base="xs:string">
-            <xs:maxLength value="3071"/>
-        </xs:restriction>
-    </xs:simpleType>
-
-    <xs:simpleType name="Uuid">
-        <xsd:restriction base="xsd:string">
-            <xsd:pattern value="([0-9]|[a-f]|[A-F]){8}-([0-9]|[a-f]|[A-F]){4}-([0-9]|[a-f]|[A-F]){4}-([0-9]|[a-f]|[A-F]){4}-([0-9]|[a-f]|[A-F]){12}"/>
-        </xsd:restriction>
-    </xs:simpleType>
-
-    <xs:simpleType name="SaslMechName">
-        <xs:restriction base="xs:string">
-            <xs:minLength value="1"/>
-            <xs:maxLength value="20"/>
-        </xs:restriction>
-    </xs:simpleType>
-
-    <xs:simpleType name="SaslData">
-        <xs:restriction base="xs:base64Binary"/>
-    </xs:simpleType>
+          </xs:complexType>
+        </xs:element>
+        <xs:element name="text" type="xs:string" minOccurs="0"/>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="next">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:any namespace='##other' minOccurs='0' maxOccurs='unbounded' processContents='lax'/>
+      </xs:sequence>
+      <xs:attribute name="task" type="SaslMechName"/>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="task-data">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:any namespace='##other' minOccurs='0' maxOccurs='unbounded' processContents='lax'/>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:simpleType name="Jid">
+    <xs:restriction base="xs:string">
+      <xs:maxLength value="3071"/>
+    </xs:restriction>
+  </xs:simpleType>
+
+  <xs:simpleType name="Uuid">
+    <xs:restriction base="xs:string">
+      <xs:pattern value="([0-9]|[a-f]|[A-F]){8}-([0-9]|[a-f]|[A-F]){4}-([0-9]|[a-f]|[A-F]){4}-([0-9]|[a-f]|[A-F]){4}-([0-9]|[a-f]|[A-F]){12}"/>
```

Comment:
Is there a historical reason that `([0-9]|[a-f]|[A-F]){8}` isn't written as `([0-9a-fA-F]){8}` ?
Maybe compatibility with certain tools?

Reply:
I probably just copy/pasted it from somewhere else, feel free to PR that change as it makes it slightly more readable. :)